dtfs-central-api/src/utils/getCrossDataModelProperty.js

```diff
@@ -0,0 +1,23 @@
+const CONSTANTS = require('../constants');
+
+const getProperty = (propertyPath) => propertyPath.split('.')[propertyPath.split('.').length - 1];
+
+const getGEFProperty = (propertyPath) => {
+  const property = getProperty(propertyPath);
+
+  switch (property) {
+    case 'supplier-name':
+      return 'dealSnapshot.exporter.companyName';
+    default:
+      return propertyPath;
+  }
+};
+
+const getCrossDataModelProperty = (deal, propertyPath) => {
+  if (deal && propertyPath) {
+    if (deal.dealSnapshot.dealType === CONSTANTS.DEALS.DEAL_TYPE.GEF) return getGEFProperty(propertyPath);
+  }
+  return propertyPath;
+};
+
+module.exports = getCrossDataModelProperty;
```

Comment:
Currently `deals-table.njk` has button values based on BSS, e.g:

```js
buttonValue: 'dealSnapshot.submissionDetails.supplier-name',
```

I was thinking it could be changed to be like the GEF model which we want to get towards:

```js
buttonValue: 'dealSnapshot.exporter.companyName',
```

Then in `getCrossDataModelProperty.js`, could do the mapping so if it's BSS, use `submissionDetails.supplier-name` instead

This way when the data models are aligned, it can just be removed from this one function. Also wouldn't need the `getProperty` function. What do you think?

Also because this is doing mapping I think it should go into a mappings directory/file instead of utils?

Reply:
👍 no worries, GEF is the newer cleaner version - very flat, whereas BSS has lots of nesting and some strange/bad field names